src/ceph-volume/ceph_volume/systemd/main.py

```diff
@@ -94,8 +118,33 @@ def main(args=None):
 
     tries = os.environ.get('CEPH_VOLUME_SYSTEMD_TRIES', 30)
     interval = os.environ.get('CEPH_VOLUME_SYSTEMD_INTERVAL', 5)
+
+    if sub_command == 'lvm':
+        block_volume = get_block_volume(extra_data)
+        if block_volume:
+            wal_device = block_volume.tags.get('ceph.wal_device')
+            db_device = block_volume.tags.get('ceph.db_device')
+
     while tries > 0:
         try:
+            # Waiting for WAL/DB availability
+            if wal_device:
+                wal_volume = get_wal_volume(wal_device)
+                if not wal_volume:
+                    logger.warning('failed to find wal volume %s, retries left: %s', wal_device, tries)
+                    tries -= 1
+                    time.sleep(interval)
```

Comment:
Seems like this is just adding additional waiting time to find the WAL or BLOCK. If 30 tries at 5 second interval is not enough for a single OSD, why can't this be a configuration change to the variables instead of adding code? 

We suspected we would face some systems that may experience delays, which is why we didn't hard-code the tries and intervals, and made them configurable instead

Reply:
That's not the role of environment variables CEPH_VOLUME_SYSTEMD_TRIES and CEPH_VOLUME_SYSTEMD_INTERVAL ?

Reply:
@alfredodeza As I understand the original ticket the issue is that the command does succeed without WAL or BLOCK being available, so the retries are not effective. @dcasier is that correct?

Reply:
ok, i just understand the question: in the original ticket, ceph-volume command does not fail if WAL and DB are not available (but the osd crash). 
The loop is not replayed

Reply:
So the problem then needs to be solved for the call at line 152 (`process.run(command, terminal_logging=False)`) vs. adding a loop here.

The activation happening in `command` needs to raise the `RuntimeError` so that none of these extra loops are needed